tests/containers/privileged_mode.pm

```diff
@@ -0,0 +1,52 @@
+# SUSE's openQA tests
+#
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: podman
+# Summary: Test podman with systemd
+# Maintainer: qa-c@suse.de
+
+use Mojo::Base 'containers::basetest';
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use utils qw(validate_script_output_retry);
+use containers::utils qw(reset_container_network_if_needed);
+
+sub run {
+    my ($self, $args) = @_;
+    select_serial_terminal;
+
+    my $runtime = $args->{runtime};
+    my $engine = $self->containers_factory($runtime);
+    $self->{runtime} = $engine;
+    reset_container_network_if_needed($runtime);
+
+    my $image = get_var("CONTAINER_IMAGE_TO_TEST", "registry.suse.com/bci/bci-base:latest");
+
+    record_info('Test', 'Launch a container with privileged mode');
+    assert_script_run("$runtime run -ti --rm --privileged $image ls /dev");
+    assert_script_run("$runtime run -ti --rm --privileged $image ls /proc/1/ns/");
+    assert_script_run("$runtime run -ti --rm --privileged $image mount -t tmpfs none /mnt");
```

Comment:
```suggestion
    # Mounting tmpfs only works in privileged mode because the read-only protection in the default mode
    assert_script_run("$runtime run -ti --rm --privileged $image mount -t tmpfs none /mnt");
```

Reply:
do you mean to explain all cases? ok